src/ServiceControl.LoadTests.QueueLengthMonitor/MsmqExtensions.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Messaging;
+using System.Runtime.InteropServices;
+
+static class MsmqExtensions
+{
+    /// <remarks>
+    /// Source: http://functionalflow.co.uk/blog/2008/08/27/counting-the-number-of-messages-in-a-message-queue-in/
+    /// </remarks>
+    public static int GetCount(this MessageQueue self)
+    {
+        var props = new Win32.MQMGMTPROPS { cProp = 1 };
+        try
+        {
+            props.aPropID = Marshal.AllocHGlobal(sizeof(int));
```

Comment:
Ahh, the magic of native MSMQ ritual invocations...

![magic](https://media.giphy.com/media/13LpnNjQU9tCtW/giphy.gif)